### PR TITLE
feat(Button): refactor to flex and center icons

### PR DESCRIPTION
**Pull request by teleginzhenya**
Button component was refactored to use modern flex. Position absolute and complex padding calculations was removed.

### CODE REVIEW

File: src/components/Button/Button.scss [hunk 1]
```diff
@@ -266,49 +255,27 @@ $block: '.#{variables.$ns}button';
...
-        position: relative;
-        width: var(--g-button-height, var(--_--height));
+        display: inline-flex;
+        justify-content: center;
+        width: var(--g-button-height, var(--_--icon-size));
         height: var(--g-button-height, var(--_--height));
         pointer-events: none;
 
-        &::after {
-            content: '\00a0';
```
Comment (teleginzhenya) [inline src/components/Button/Button.scss:280]: Seems like it didn't have any effect before

**Review by teleginzhenya — COMMENTED** [review]
And seems like we might need to add some tests/stories to the button

* custom width
* max width

File: src/components/Button/Button.scss [hunk 2]
```diff
@@ -12,24 +12,9 @@ $block: '.#{variables.$ns}button';
...
-                        --g-button-icon-size,
-                        var(--_--icon-size)
-                    )
-            ) / 2
-    );
     --_--font-size: var(--g-text-body-1-font-size);
 
     @include mixins.button-reset();
-    display: inline-block;
-    position: relative;
```
Comment (amje) [inline src/components/Button/Button.scss:32]: There are children elements with `absolute` position (:before, :after), this prop can't be omitted

File: src/components/Button/Button.scss (hunk 1 above)
Comment (amje) [reply src/components/Button/Button.scss:280]: Please rebase your branch, I added test with text interation.

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.scss (hunk 2 above)
Comment (teleginzhenya) [reply src/components/Button/Button.scss:32]: fixed 79ccb7a622dc73a312669206b7b58c3ab85dc9e9

File: src/components/Button/Button.scss (hunk 1 above)
Comment (teleginzhenya) [reply src/components/Button/Button.scss]: returned and fixed 85ec4446b414efc7d2038ff8efb5307cb5914587

File: src/components/Button/Button.scss [hunk 3]
```diff
@@ -266,13 +256,12 @@ $block: '.#{variables.$ns}button';
...
         display: inline-block;
         white-space: nowrap;
-        padding: 0 var(--g-button-padding, var(--_--padding));
     }
 
     &__icon {
         display: inline-block;
         position: relative;
-        width: var(--g-button-height, var(--_--height));
+        width: var(--g-button-height, var(--_--icon-size));
```
[the remark this answers is not in the document]
Comment (teleginzhenya) [reply src/components/Button/Button.scss]: Fixed here b269fd5886173fdb1c40916559edcb9d09613e27 and I've added some screenshot tests for this scenario in [separate pull request](https://github.com/gravity-ui/uikit/pull/1455)

File: src/components/Button/Button.scss [hunk 4]
```diff
@@ -289,26 +287,12 @@ $block: '.#{variables.$ns}button';
...
+            order: 0;
         }
 
         &_side_end {
-            inset-inline-end: var(--_--icon-position);
-
-            & ~ #{$block}__text {
-                padding-inline-end: var(--_--icon-space);
-            }
+            order: 2;
```
Comment (amje) [inline src/components/Button/Button.scss:295]: Default value for order is `0`, let's use`-1` and `1` then?

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.scss (hunk 4 above)
Comment (teleginzhenya) [reply src/components/Button/Button.scss]: 4490b68a4c8192e11597fcf3a6044d164705149a

**Pull request merged**